src/react/types.ts

```diff
@@ -18,8 +20,16 @@ export interface UseBaseQueryOptions<
 export interface UseQueryOptions<
   TQueryFnData = unknown,
   TError = unknown,
-  TData = TQueryFnData
-> extends UseBaseQueryOptions<TQueryFnData, TError, TData> {}
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey
+>
+  extends Omit<
+    UseBaseQueryOptions<TQueryFnData, TError, TData>,
+    'queryFn' | 'queryKey'
+  > {
+  queryKey?: TQueryKey
+  queryFn?: QueryFunction<TQueryFnData, TQueryKey>
+}
 
 export interface UseInfiniteQueryOptions<
```

Comment:
could you maybe also add the support for infinite queries as well?

Reply:
I could in the future, but I think I would prefer to do that on a separate branch (and perhaps another one for useMutation?).